scripts/breaking_changes_checker/breaking_changes_tracker.py

```diff
@@ -183,24 +192,49 @@ def run_non_breaking_class_level_diff_checks(self, module: Dict) -> None:
-                                    self.module_name, self.class_name, method_name
-                                )
-                                self.features_added.append(fa)
+                        if not isinstance(self.function_name, jsondiff.Symbol):
+                            if self.function_name not in stable_methods_node:
+                                if self.class_name.endswith("Client"):
+                                    # This is a new client method
+                                    fa = (
+                                        self.ADDED_CLIENT_METHOD_MSG,
+                                        ChangeType.ADDED_CLIENT_METHOD,
+                                        self.module_name, self.class_name, method_name
+                                    )
+                                    self.features_added.append(fa)
+                                else:
+                                    # This is a new class method
+                                    fa = (
+                                        self.ADDED_CLASS_METHOD_MSG,
+                                        ChangeType.ADDED_CLASS_METHOD,
+                                        self.module_name, class_name, method_name
+                                    )
+                                    self.features_added.append(fa)
                             else:
-                                # This is a new class method
-                                fa = (
-                                    self.ADDED_CLASS_METHOD_MSG,
-                                    ChangeType.ADDED_CLASS_METHOD,
-                                    self.module_name, class_name, method_name
-                                )
-                                self.features_added.append(fa)
+                                # Check existing methods for new parameters
+                                stable_parameters_node = stable_methods_node[self.function_name]["parameters"]
+                                current_parameters_node = self.current[self.module_name]["class_nodes"][self.class_name]["methods"][self.function_name]["parameters"]
+                                for param_name, param_components in method_components.get("parameters", {}).items():
+                                    self.parameter_name = param_name
+                                    if self.parameter_name not in stable_parameters_node and \
+                                            not isinstance(self.parameter_name, jsondiff.Symbol):
+                                        if self.function_name == "__init__":
+                                            # If this is a new class property skip reporting it here and let the class properties check handle it
+                                            if self.parameter_name in class_components.get("properties", {}).keys():
+                                                continue
```

Comment:
I think I got confused here - the comment talks about new class property, but aren't we looking at parameters for the init here?

Reply:
Yes however if we add new parameters in the init corresponding to new class properties we dont want to report that twice, hence doing some clean up here to only report new class properties once. For now, I think it's reasonable to only report new properties, but if we get feedback that it would be better to also report the corresponding parameter in the init we can change this as needed.

Reply:
The changelog report looks redundant if we leave both. The original changelog reporter only reported this once.

Reply:
I see, maybe leave a comment in the code explaining this